views/overlays/matchIntro.hbs

```diff
@@ -27,91 +35,148 @@
+			  					elem.show();
+			  				});
+  						}, 25);				
+  					}, 300);
+  				}, 550);
+  			}, 100);			
+  		}
 
-        if(data.title != document.getElementById('title').innerText) {
-          $('#title').trigger('textChanged', data.title || '');
-        }
+  		// For some reason the autoplay property doesn't work in Chrome
+  		document.getElementById("audio-chime").pause();
 
-        if(data.lplayer != document.getElementById('lplayer').innerText) {
-          $('#lplayer').trigger('textChanged', data.lplayer || '');
-        }
+  		var initialBackground = 'url(http://i.imgur.com/tlhS4vA.png)';
+  		var step = 0;
+      var socket = io('/overlay');
 
-        if(data.rplayer != document.getElementById('rplayer').innerText) {
-          $('#rplayer').trigger('textChanged', data.rplayer || '');
-        }
+      socket.on('play intro', function(data) {
+      	if(data.flag !== step) { 
+      		return true;
+      	}
 
-        document.getElementById("leftCharacter").src = charImageDict[data.lCharacter];
-        document.getElementById("rightCharacter").src = charImageDict[data.rCharacter];
-      });
+      	var showArr = [$('#title'), $('#lplayer'), $('#rplayer')];
 
-		$('.text').bind('textChanged', function(e, newText){
-	        $(this).animate({'opacity': '0', 'margin-left': '-1%'}, 350, function() {
-	            $(this).text(newText);
-	          }).animate({'opacity': '1', 'margin-left': '0'}, 350);
-	      });
-	    });
+      	if (data.flag === 0) {
+        	// Reset the page and fade out background
+        	$('#container').animate({'opacity': '0'}, 500, function () {
+        		  $('#container').css('background', initialBackground);
+        			$('#lCharacter').css({'opacity': '0', 'margin-left': '-1%'});
+		        	$('#rCharacter').css({'opacity': '0', 'margin-left': '-1%'});    	
+		        	$('#vs').css('font-size', '100em').hide();
+		        	showArr.forEach(function(elem) {
+		        		elem.hide();
+		        	});
+        		});
 
+        		step = 1;
+      	} else if (data.flag === 1) {
+      		// Update the page with all relevant data/images
+          $('#title').text(data.title || '');
+          $('#lplayer').text(data.lplayer || '');
+          $('#rplayer').text(data.rplayer || '');
+          $('#lCharacter').attr('src', charImageDict[data.lCharacter] || '');
+          $('#rCharacter').attr('src', charImageDict[data.rCharacter] || '');
+
+          step = 2;
+				} else {
+      		// Show the background again and begin animation
+          $('#container').animate({'opacity': '1'}, 500, function () {
+          	setTimeout(function() {
+							$('#lCharacter').trigger('fadeIn');
+							setTimeout(function () {
+								$('#rCharacter').trigger('fadeIn');
+		          	setTimeout(function () {
+		          		transitionBackground(showArr);
+		          	}, 500); 
+	          	}, 500); 
+	      		}, 100);  
+          }); 
+
+          step = 0;  		       
+      	}     
+      });
```

Comment:
A lot of this whitespace looks really wonky. Could you standardize it? It might look fine in Sublime/whatever editor, but Github will interpret things literally.